src/main/java/com/gluonhq/richtext/RichTextAreaSkin.java

```diff
@@ -97,6 +98,7 @@ protected RichTextAreaSkin(final RichTextArea control) {
         // all listeners have to be removed within dispose method
         control.editableProperty().addListener(this::editableChangeListener);
         editableChangeListener(null); // sets up all related listeners
+        control.textLengthProperty.bind(viewModel.textLengthProperty());
```

Comment:
Use the method. The field `control.textLengthProperty` should be private? 
Or you are making it package protected so you can bind it from skin?

That can break if we decide to introduce a skin package...

It's probably a better approach to add a listener, that can be disposed as the rest of the listeners.

Reply:
I think we still have the same issue since control's `textLengthProperty` is readonly, so setter has to be package protected. Is there a better solution?

Reply:
Yes, you are right, that won't work.

At least with the typical MVC approach, the control has some properties and the skin listens to them, but not the other way around. The skin could be removed after all to use a different implementation, so we shouldn't rely on it to update a control property.

Also the control should have a `textProperty` that the user can set with some initial text, and that should go to the skin, which should update the ModelView with it.

After every update of that text, the `textProperty` should have a face value of the current pieceTable. That would allow the user to serialize the "final" text to a file for instance. As a consequence of changes in this property, the lengthTextProperty in the control should not probably needed at all.

With that textProperty, we still need to find a way to update the control from any change in the textBuffer, but it is not clear to me if with MVVC the control should see the model at all.

Reply:
Anyway, this is probably for a follow-up. Once you are happy with the changes, lets merge and restart the discussion.